files/default/lib/bumpzone.rb

```diff
@@ -0,0 +1,115 @@
+#!/opt/chef/embedded/bin/ruby
+require 'git'
+require 'json'
+require 'octokit'
+
+# Library to bump DNS zones
+class BumpZone
+  @commit_msg = "\n\n"
+  @warning_msg = "Warning:\n\n"
+  @warning = false
+  ONELINE_REGEX = /([\w\d\.-_]+\s+[\d]+\s+IN\s+SOA.*[\w\d\.-_]+\s+)(\d{10})(\s+[\d]+\s+[\d]+\s+[\d]+\s+[\d]+)/
+  MULTILINE_REGEX = /([\w\d\.-_]+\s+IN\s+SOA\s+[\w\d\.-_]+\s+[\w\d\.-_]+\s+\(\n\s+)(\d{10})(.*\n.*\n.*\n.*\n.*\n\s+\))/
```

Comment:
So, given that staff will eventually writing and changing zone files directly, there's a couple of problems with these regex:

 [ ] trailing whitespace after ( in an SOA record is not recognized
 [ ] hyphens in domains are not recognized (`-` is treated as a range operator within `[]` unless escaped)
 [ ] comments are not represented -- a commented out SOA record would be matched, parsed and updated